packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx

```diff
@@ -0,0 +1,313 @@
+import Switch from '@/ds-components/Switch';
+import TextInput from '@/ds-components/TextInput';
+import useApi from '@/hooks/use-api';
+import useTenantPathname from '@/hooks/use-tenant-pathname';
+import { trySubmitSafe } from '@/utils/form';
+import { uriValidator } from '@/utils/validator';
+
+import styles from './index.module.scss';
+import {
+  buildIdpInitiatedAuthConfigEndpoint,
+  type IdpInitiatedAuthConfigFormData,
+  parseResponseToFormData,
+  parseFormDataToRequestPayload,
+} from './utils';
+
+type FormProps = {
+  readonly ssoConnector: SsoConnectorWithProviderConfig;
+  readonly applications: Application[];
+  readonly idpInitiatedAuthConfig: SsoConnectorIdpInitiatedAuthConfig | undefined;
+  readonly mutateIdpInitiatedConfig: KeyedMutator<SsoConnectorIdpInitiatedAuthConfig>;
+};
+
+function ConfigForm({
+  ssoConnector,
+  applications,
+  idpInitiatedAuthConfig,
+  mutateIdpInitiatedConfig,
+}: FormProps) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const { getTo } = useTenantPathname();
+  const api = useApi();
+
+  const {
+    control,
+    register,
+    formState: { isDirty, isSubmitting, errors },
+    reset,
+    setValue,
+    handleSubmit,
+    watch,
+  } = useForm<IdpInitiatedAuthConfigFormData>({
+    defaultValues: parseResponseToFormData(idpInitiatedAuthConfig),
+  });
+
+  const isIdpInitiatedSsoEnabled = watch('isIdpInitiatedSsoEnabled');
+  const defaultApplicationId = watch('config.defaultApplicationId');
+  const autoSendAuthorizationRequest = watch('config.autoSendAuthorizationRequest');
+
+  const defaultApplication = useMemo(() => {
+    return applications.find((application) => application.id === defaultApplicationId);
+  }, [applications, defaultApplicationId]);
+
+  const emptyApplicationsError = useMemo<ReactElement | undefined>(() => {
+    if (applications.length === 0) {
+      return (
+        <Trans
+          components={{
+            a: <Link className={styles.inlineLink} to={getTo('/applications')} />,
+          }}
+        >
+          {t('enterprise_sso_details.idp_initiated_auth_config.empty_applications_error')}
+        </Trans>
+      );
+    }
+  }, [applications, getTo, t]);
+
+  const defaultApplicationRedirectUris = useMemo(
+    () => defaultApplication?.oidcClientMetadata.redirectUris ?? [],
+    [defaultApplication]
+  );
+
+  const emptyRedirectUrisError = useMemo(() => {
+    if (defaultApplication && defaultApplicationRedirectUris.length === 0) {
+      return t('enterprise_sso_details.idp_initiated_auth_config.empty_redirect_uris_error');
+    }
+  }, [defaultApplication, defaultApplicationRedirectUris.length, t]);
+
+  // Force set autoSendAuthorizationRequest to false if the default application is set to SPA
+  useEffect(() => {
+    if (defaultApplication?.type === ApplicationType.SPA) {
+      setValue('config.autoSendAuthorizationRequest', false);
+    }
+  }, [defaultApplication?.type, setValue]);
```

Comment:
This `useEffect` can be saved by writting the `setValue` in the `onChange` function, IMO

Reply:
`defaultApplication` is a computed value. Using `useEffect` to subscribe the changes is more straightforward.